From fdedd87a135261e41ed790b04f4dc085c132285c Mon Sep 17 00:00:00 2001
From: Harish Mahendrakar <harish.mahendrakar@ittiam.com>
Date: Thu, 5 Oct 2017 14:18:26 +0530
Subject: [PATCH 2/8] SoftAVCDec: Handle zero length input without EOS

When a zero length input buffer is sent to SoftAVC decoder
without EOS set in nFlags, SoftAVC decoder plugin was
entering an infinite loop. Fixed it by returning from
onQueueFilled for such cases.

Bug: 66969349
Test: Tested using poc associated with the bug
Change-Id: I79cebc8f5b40c51256aba83a08deb547b220c4d7
(cherry picked from commit 21943c05f0ee2350647a6fa5ee17aa3c68859383)
---
 media/libstagefright/codecs/avcdec/SoftAVCDec.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/libstagefright/codecs/avcdec/SoftAVCDec.cpp b/media/libstagefright/codecs/avcdec/SoftAVCDec.cpp
index cecc52b..1f47155 100644
--- a/media/libstagefright/codecs/avcdec/SoftAVCDec.cpp
+++ b/media/libstagefright/codecs/avcdec/SoftAVCDec.cpp
@@ -578,7 +578,7 @@ void SoftAVC::onQueueFilled(OMX_U32 portIndex) {
                 notifyEmptyBufferDone(inHeader);
 
                 if (!(inHeader->nFlags & OMX_BUFFERFLAG_EOS)) {
-                    continue;
+                    return;
                 }
 
                 mReceivedEOS = true;
-- 
2.7.4

